helene/reqs.md

```diff
@@ -1,49 +1,129 @@
+
+
+# General philosophy/concepts
+
+## For cleaner/better code
   - never override an CSS declaration
   - keep our mixins simple, you shouldn't need a map and two days to find where that mixin was created  
   - from general to specific, keeping it as generic as possible
-  
-###For an ideal slots/skeleton framework
+
+## For an ideal slots/skeleton framework
   - we add, we never remove (avoid regression testing)
   - the framework is versioned, as with an API
   - developers should never have to look at the skeleton
+
+---
+
+# Phil's Stories
+
+# CSS Stories
+
+As a **new Developer** I want to...
+
+1. know how to make my first contribution so I can contribute quickly
+
+
+As a **Developer** I want to...
+
+1. change files and run tests easily so I am confident I didn't break anything with my change
+  - _(naive styling test: pixel-compare each PDF (just needs to verify nothing changed) )_
+  - _(naive collation test: generate the HTML & diff)_
+1. take an existing book and quickly add a new "feature"
+1. take an existing book and change some collation to make a new book
+1. quickly make a new book from scratch
+1. generate HTML from parts of books or entire books (with caching a-la-makefile) easily from the commandline
+1. have "unit tests" for the CSS files so I don't have to generate the whole book
+  - _(also acts as "runnable documentation")_
+1. maintain as little documentation as possible because keeping documentation in sync is error-prone
+  - _(including scripts to generate the HTML, installing dependencies, etc. (ie [Runnable Documentation](http://githubengineering.com/runnable-documentation/)) )_
+1. associate the CSS files for books with the code somewhere in GitHub and keep them in sync
+  - _(:cake: commit hooks which republish?)_
+1. an option to add debugging information to the output to see where things came from
+  - _(:cake: including sourcemap into the CSS file maybe :smile:)_
+1. have the passes to be abstracted away because most of those will not change frequently
+1. write simple CSS as long as I know what the mixins are and create few (if any) of my own mixins.
+  - (:thought_balloon: The analysis done by kerwin+derek(?) should contain all the mixins)
+
+
+As an **External user** I want to...
+
+1. easily tweak the default numbering/collation for my book because I may want to refer to things differently
+
+
+As a **Translator** I want to...
+
+1. easily change strings and numbering because languages have different conventions
+1. easily change LTR to RTL
+
+
+As **openstax** I want to...
+
+1. have strings & numbering to work in multiple languages because a lot of the content is not in english
+1. have a default numbering/collation for non-openstax books
+  - _(maybe also affects the cnxml2html conversion for things like exercises (they'll need to not lose some metadata))_
+1. make sure people cannot create infinite loops
```

Comment:
This should be under validation, I think. Validation of rules should be enforced for everyone, not just external user.